app/models/concerns/users/roles.rb

```diff
@@ -86,6 +86,10 @@ def roles_has_changed?
     roles_changed || organization_roles.any?(&:changed?)
   end
 
+  def has_supreme_user?
+   root if root && user != root.user
```

Comment:
```suggestion
   root unless root == self
```
Debería funcionar igual y se entiende mejor.